main.py

```diff
@@ -193,16 +194,23 @@ def update_circuits(self):
     def do_match(cls, flow, args):
         """Match a packet against this flow (OF1.3)."""
         # pylint: disable=consider-using-dict-items
+        # pylint: disable=too-many-return-statements
         if ('match' not in flow['flow']) or (len(flow['flow']['match']) == 0):
             return False
         for name in flow['flow']['match']:
             field_flow = flow['flow']['match'][name]
+            if name == 'dl_vlan':
+                if name not in args:
+                    if field_flow != 0:
```

Comment:
That's correct, if `dl_vlan` doesn't exist on `args` and if field_flow != 0, then that's a mismatch. However, for this case the arg `a` aka `field` should be passed as `None`. Can you move this statement as a early check into `match_field_dl_vlan`? That also improves testability, as it is it's leaking matching `dl_vlan` logic outside of `match_field_dl_vlan`

Reply:
Fine @viniarck, I agree.